### DISCUSSION TITLE
Provisioning through TCP

### DISCUSSION THREAD

**Discussion by sojan1** [Mimetis/Dotmim.Sync#1087] [Q&A] [answered]
1. Provisioning through TCP is working well. 
2. Client can synchronize the data through the api (http://192.168.1.77:81/api/sync*) after Provisioning through TCP
3. But the **filter parameters are ignored** when synchronized after TCP Provisioning and it **brings every record** from the table.

![image](https://github.com/Mimetis/Dotmim.Sync/assets/111710684/fa9b694b-07b6-4218-9e1b-1d5792f71991)

Is there any extra settings required for filters, when provisioned through TCP with no synchronisation?. 

The same code works perfectly when provisioning happens automatically with synchronisation through http://192.168.1.77:81/api/sync. 
![image](https://github.com/Mimetis/Dotmim.Sync/assets/111710684/ed3e9450-cecd-43a7-a79b-53950049a016)

**Client Device**
 
    var serverOrchestrator = new WebRemoteOrchestrator("http://192.168.1.77:81/api/sync");
    var clientProvider = new SqliteSyncProvider(this.settings.DataSource);
    var options = new SyncOptions
    { BatchDirectory = Path.Combine(SyncOptions.GetDefaultUserBatchDirectory(), "client") };
    var agent = new SyncAgent(clientProvider, serverOrchestrator, options);
    var progress = new SynchronousProgress<ProgressArgs>(pa => Console.WriteLine($"{pa.ProgressPercentage:p}\t {pa.Message}"));
    var parameters = new SyncParameters
                {
                    { "CLNT_ID","D7" }
                };

    var s1 = await agent.SynchronizeAsync("v0", parameters, progress);
    Console.WriteLine(s1);

**Windows Provisioning** -  On ButtonClick 

        SqlSyncProvider serverProvider = new SqlSyncProvider(
          @"Data Source=SERVERMachine\SQLEXPRESS01; Initial Catalog=CORETEST; Integrated Security=true;Trusted_Connection=true;TrustServerCertificate=True;MultipleActiveResultSets=False;");
        var progress = new SynchronousProgress<ProgressArgs>(pa => Console.WriteLine($"{pa.ProgressPercentage:p}\t {pa.Message}"));
        var options = new SyncOptions
        {BatchDirectory = Path.Combine(SyncOptions.GetDefaultUserBatchDirectory(), "server") };
        var remoteOrchestrator = new RemoteOrchestrator(serverProvider);
        var setup = new SyncSetup("_clnt");
        await remoteOrchestrator.ProvisionAsync("v0", setup); 
i am using the same reference on the windows application too.
 

**Code on Proxy** - **http://192.168.1.77:81/api/sync**

           var connectionString = builder.Configuration.GetSection("ConnectionStrings")["SqlConnection"];
            var options = new SyncOptions
            { BatchDirectory = Path.Combine(SyncOptions.GetDefaultUserBatchDirectory(), "server") };
 
            var setup = new SyncSetup("_clnt", "PROJECT");
            var clntFilter = new SetupFilter("_clnt");
            clntFilter.AddParameter("CLNT_ID", "_clnt", true);
            clntFilter.AddWhere("CLNT_ID", "_clnt", "CLNT_ID");
            setup.Filters.Add(clntFilter);

            var projFilter = new SetupFilter("PROJECT");
            projFilter.AddParameter("CLNT_ID", "PROJECT", true);
            projFilter.AddWhere("CLNT_ID", "PROJECT", "CLNT_ID");
            setup.Filters.Add(projFilter);
            builder.Services.AddSyncServer<SqlSyncProvider>(connectionString, setup,null,null,"v0");

**Comment by Mimetis** [comment, marked as answer]
I dont see filter declaration in your windows provisioning 

``` cs
 SqlSyncProvider serverProvider = new SqlSyncProvider(
      @"Data Source=SERVERMachine\SQLEXPRESS01; Initial Catalog=CORETEST; Integrated Security=true;Trusted_Connection=true;TrustServerCertificate=True;MultipleActiveResultSets=False;");
    var progress = new SynchronousProgress<ProgressArgs>(pa => Console.WriteLine($"{pa.ProgressPercentage:p}\t {pa.Message}"));
    var options = new SyncOptions
    {BatchDirectory = Path.Combine(SyncOptions.GetDefaultUserBatchDirectory(), "server") };
    var remoteOrchestrator = new RemoteOrchestrator(serverProvider);
    var setup = new SyncSetup("_clnt");
    await remoteOrchestrator.ProvisionAsync("v0", setup); 
```
In your proxy

``` cs
 var connectionString = builder.Configuration.GetSection("ConnectionStrings")["SqlConnection"];
        var options = new SyncOptions
        { BatchDirectory = Path.Combine(SyncOptions.GetDefaultUserBatchDirectory(), "server") };

        var setup = new SyncSetup("_clnt", "PROJECT");
        var clntFilter = new SetupFilter("_clnt");
        clntFilter.AddParameter("CLNT_ID", "_clnt", true);
        clntFilter.AddWhere("CLNT_ID", "_clnt", "CLNT_ID");
        setup.Filters.Add(clntFilter);

        var projFilter = new SetupFilter("PROJECT");
        projFilter.AddParameter("CLNT_ID", "PROJECT", true);
        projFilter.AddWhere("CLNT_ID", "PROJECT", "CLNT_ID");
        setup.Filters.Add(projFilter);
        builder.Services.AddSyncServer<SqlSyncProvider>(connectionString, setup,null,null,"v0");
```

You are declaring a scope "v0" with two tables and filters, but in your windows application you are declaring only one table without filter with the same scope name "v0"
That's not correct

What is the purpose of your windows application ? Provisioning the server ? Or provisioning your clients ?
Why do you have two setups, one with 1 tables and one with 2 tables ?
Why these two setups are saved with the same scope name ?

**Comment by sojan1** [comment]
I thought the filter will be handled on web-api as the client's request comes through the web-api. http://192.168.1.77:81/api/sync and thus i did not filter when provisioned through the windows.

1. The purpose of windows application is to do the provisioning the server. what client does is only synchronisation.  Provisioning tool will do the versioning too.
2. i have added the second table to windows provisioning. Should i also add the filter parameters to the windows provisioning?
3. I want the scope created on windows to be used by the devices..

**Comment by sojan1** [comment]
i think it works now.  **Thank you very much.**
1. i have added the filter parameters to the windows application same like what i have in the web-api
2. added the PROJECT table too.
